api/client/pull.go

```diff
@@ -59,7 +59,7 @@ func (cli *DockerCli) CmdPull(args ...string) error {
 	authConfig := cli.resolveAuthConfig(repoInfo.Index)
 	requestPrivilege := cli.registryAuthenticationPrivilegedFunc(repoInfo.Index, "pull")
 
-	if isTrusted() && !ref.HasDigest() {
+	if trust_should_lookup(ref, repoInfo.Official) {
```

Comment:
Should this `ref` argument be `ref.HasDigest()`? Not sure how `ref` can be passed as a boolean.